### PR TITLE
Atrium

### LINKED ISSUE #36: Sample wanted for robstoll/atrium
## About the library

- *Description*:    A multiplatform assertion library for Kotlin 
- *GitHub*: https://github.com/robstoll/atrium
- *Official website*: https://docs.atriumlib.org/

## What the sample could do

Add kluent examples in src/test, maybe follow what was done for strikt
https://github.com/LouisCAD/kotlin-libraries-playground/tree/main/kotlin-jvm/src/test/kotlin/framework
See also kluent https://github.com/LouisCAD/kotlin-libraries-playground/issues/35

## Wants to contribute the sample?

- Say so in a comment here
- Follow instructions in [CONTRIBUTING.md](https://github.com/LouisCAD/kotlin-libraries-playground/blob/main/CONTRIBUTING.md)
- Submit your pull request!

### PR DESCRIPTION
**Pull request by oppongjr**
Closes #36

### CODE REVIEW

**Comment by jmfayard** [comment]
Hello Isaac, 
I don't think the `DistanceConverterParameterizedTest` is needed at all here. 
We have it for junit5 and kotest because both of those frameworks have their own way to do parameterized tests.
But Atrium concerns itself only with assertions.

So I would rather have only the other file, but with more examples of Atrium assertions

**Comment by oppongjr** [comment]
Okay, I see what you mean. I'll update the PR with more assertions. .

**Comment by jmfayard** [comment]
Notes:

The testing samples will be in a `kotlin-testing` module once this is merged https://github.com/LouisCAD/kotlin-libraries-playground/pull/49

Usually we want all tests to pass, but here since we are testing the testing frameworks, we want to show what they do when the tests fails, so some of the assertions that you write should be failing!

**Comment by oppongjr** [comment]
Very well noted. Thank you!

**Comment by jmfayard** [comment]
Hello @oppongjr How is it going? Are you blocked somewhere?

**Comment by oppongjr** [comment]
@jmfayard Not really. I've just been overwhelmed with work. I'll push an update to the branch tonight after I resolve the conflicts. You might need to help with the test function names. Naming is hard, truly.

**Comment by jmfayard** [comment]
@oppongjr no problem

**Comment by LouisCAD** [comment]
Ah yes, naming is hard.

Maybe we should add a mention in the contributing guide that the issue or draft PR is the right place to discuss naming and request help from us and people that would join the discussion? 🤔

**Comment by jmfayard** [comment]
@oppongjr I refactored the tests, because it seems that nobody understood how it was strucuted :) 
While I was at it, I solved your merge conflicts.
Please pull my changes
Then replace the content of  `kotlin-testing/src/test/kotlin/testing/asserts/AtriumAsserts.kt` with whatever you had locally.

**Comment by oppongjr** [comment]
@jmfayard Done. I am not sure about the names of the test functions, so I will be really happy if you can suggest better names. I could add more assertions too.

File: kotlin-jvm/src/test/kotlin/framework/atrium/Assertions.kt
Comment (jmfayard) [inline kotlin-jvm/src/test/kotlin/framework/atrium/Assertions.kt:1]: The method names are ok-ish.

But please 
1. move the content of this class inside the class `testing.asserts.AtriumAsserts`
2. create a copy of this class in `testing.failing.FailingAtriumAsserts` where you make the tests fail so that we can see how clear the error messages are.

File: kotlin-jvm/src/test/kotlin/framework/atrium/Assertions.kt
Comment (jmfayard) [inline kotlin-jvm/src/test/kotlin/framework/atrium/Assertions.kt:9]: this should go in a class `testing.failing.FailingAtriumAsserts`

**Comment by jmfayard** [comment]
it would be nice if you can add a few more asserts, for example:
- one more with lists
- one with maps 
- one with data-driven-testing https://github.com/robstoll/atrium/blob/v0.13.0/README.md#data-driven-testing

**Comment by oppongjr** [comment]
@jmfayard well noted. working on it

**Review by jmfayard — APPROVED** [review]

**Comment by jmfayard** [comment]
LGTM to me, let me know when you think it's ready to merge

**Comment by oppongjr** [comment]
@jmfayard if it looks good to you then that's it. It can be merged now

**Comment by jmfayard** [comment]
Done, thanks a lot!

**Pull request merged**
